app/image/gc/gc.go

```diff
@@ -10,20 +10,78 @@ import (
 	"sync"
 	"time"
 
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/tsuru/config"
 	"github.com/tsuru/tsuru/api/shutdown"
 	"github.com/tsuru/tsuru/app"
 	"github.com/tsuru/tsuru/app/image"
 	tsuruErrors "github.com/tsuru/tsuru/errors"
+	"github.com/tsuru/tsuru/event"
 	"github.com/tsuru/tsuru/log"
+	"github.com/tsuru/tsuru/permission"
 	"github.com/tsuru/tsuru/registry"
 	"github.com/tsuru/tsuru/servicemanager"
 	appTypes "github.com/tsuru/tsuru/types/app"
+	permTypes "github.com/tsuru/tsuru/types/permission"
 )
 
 const (
 	imageGCRunInterval = 5 * time.Minute
 )
 
+var (
+	gcExecutionsTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "tsuru_gc_executions_total",
+		Help: "The number of times that gc had runned by phase",
+	}, []string{"phase"})
+
+	executionDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "tsuru_gc_execution_duration_seconds",
+		Help:    "How long during the GC process",
+		Buckets: prometheus.ExponentialBuckets(0.1, 3, 10),
```

Comment:
nitpick: Upper bound is 0.1*(3^10) ~= 1.6 hours. Maybe a bit excessive. :)  What about a 2.7 factor for ~30 minutes upper bound and better resolution or less buckets?